EDU-2838: update the service health and worker health docs (#3024)
temporalio/documentation: docs/production-deployment/cloud/service-health.mdx

  - health
---

> Temporal Cloud metrics help you to monitor the Service health of your production deployments. This documentation addresses some common best practices for monitoring Service health.

Reviewer: ```suggestion
Temporal Cloud metrics help monitor the service health of production deployments.
This documentation covers best practices for monitoring service health.
```

I really want to know if we're going to stick with Docs single line sentences. I've found it helpful, and it makes localizing changes and diffs far simpler but Jwahir is not an adherent and I'm the only team member of the old regime, and I put an "abolish" item into the next (non-retro) meeting.

Author: I am genuinely just very bad at remembering to adhere to this convention and I've never used it elsewhere, so I manage to forget it every time I task switch off of this repo :(

I'm not necessarily a fan, but I also don't hate it nearly as much as I must appear to.

> ## Monitoring availability issues

Reviewer: This is a SEO decision that I believe @napcs signed off on and I know Patrick and Cully adopted. We dropped present participles to make our headlines more friendly to search.

Since @jsundai was involved in the IA work, she probably has more insight into the specifics.

In addition, we started adding the question form with reworded content using bolding after the H2, for example:

```
## Monitor availability issues

**How do I detect potential problems with the Temporal Cloud service that might affect my Workflows?**
```

Again, check with Jwahir and Brian for specifics

When you see a sudden drop in Worker resource utilization, verify whether Temporal Cloud's API is showing increased latency and error rates.

### Reference Metrics

- [temporal\_cloud\_v0\_service\_latency\_bucket](/production-deployment/cloud/metrics/reference#temporal_cloud_v0_service_latency_bucket)

This metric measures latency for `SignalWithStartWorkflowExecution`, `SignalWorkflowExecution`, `StartWorkflowExecution` operations.

> These are all considered mission critical operations and are [never throttled](/cloud/service-availability#throughput), so this metric should be a good indicator of your lowest possible latency.

Reviewer: ```suggestion
These operations are mission critical and never [throttled](/cloud/service-availability#throughput).
This metric is a good indicator of your lowest possible latency.
```

### Prometheus Query for this Metric

P99 service lag (histogram):

```
histogram_quantile(0.99, sum(rate(temporal_cloud_v0_service_latency_bucket[$__rate_interval])) by (temporal_namespace, operation, le))
```

> ## Monitoring Temporal Service errors

Reviewer: Present participle. Get ruling on this.
